hack/update-deps.sh

```diff
@@ -32,8 +32,8 @@ VERSION="master"
 # The list of dependencies that we track at HEAD and periodically
 # float forward in this repository.
 FLOATING_DEPS=(
-  "knative.dev/test-infra@${VERSION}"
-  "knative.dev/pkg@${VERSION}"
+  "knative.dev/test-infra@release-0.15"
+  "knative.dev/pkg@release-0.15"
   "knative.dev/caching@${VERSION}"
```

Comment:
Nit
```suggestion
  "knative.dev/test-infra@${VERSION}"
  "knative.dev/pkg@${VERSION}"
  "knative.dev/caching@master"
```
And update VERSION above

Reply:
I discussed precisely this with @mattmoor :joy:. Changing the inline versions is easier to undo via `sed` later and most of the repos will have a 2 stage approach anyway (bumping pkg now and serving later).

Reply:
Ack, makes sense to me

Reply:
Yeah, the `sed` has lower arity this way around 😅